crates/miden-lib/asm/kernels/transaction/lib/memory.masm

```diff
@@ -1100,6 +1127,72 @@ export.get_acct_storage_slots_section_ptr
     exec.get_current_account_data_ptr push.ACCT_STORAGE_SLOTS_SECTION_OFFSET add
 end
 
+#! Returns the memory pointer to the native account's storage slots section.
+#!
+#! Inputs:  []
+#! Outputs: [storage_slots_section_ptr]
+#!
+#! Where:
+#! - storage_slots_section_ptr is the memory pointer to the native account's storage slots section.
+export.get_native_account_storage_slots_ptr
+    exec.get_native_account_data_ptr add.ACCT_STORAGE_SLOTS_SECTION_OFFSET
+end
+
+#! Returns the memory pointer to the initial storage slots of the native account.
+#!
+#! Inputs:  []
+#! Outputs: [account_initial_storage_slots_ptr]
+#!
+#! Where:
+#! - account_initial_storage_slots_ptr is the memory pointer to the initial storage slot values.
+export.get_native_account_initial_storage_slots_ptr
+  exec.get_native_account_data_ptr add.ACCT_INITIAL_STORAGE_SLOTS_SECTION_OFFSET
+end
+
+### ACCOUNT DELTA #################################################
+
+#! Increments the delta of the account's nonce.
+#!
+#! Example: If this procedure is called with nonce_delta 3 and again with nonce_delta 2, then the
+#! stored nonce_delta will be 5.
+#!
+#! Inputs:  [nonce_delta]
+#! Outputs: []
+#!
+#! Where:
+#! - nonce_delta is the value by which the nonce changed.
+export.incr_account_nonce_delta
+    exec.get_account_nonce_delta add mem_store.ACCOUNT_DELTA_NONCE_PTR
+end
+
+#! Returns the delta of the account's nonce.
+#!
+#! Inputs:  []
+#! Outputs: [nonce_delta]
+#!
+#! Where:
+#! - nonce_delta is the value by which the nonce changed.
+export.get_account_nonce_delta
+    mem_load.ACCOUNT_DELTA_NONCE_PTR
+end
+
+#! Initializes the storage slots holding the initial values of each slot.
+#!
+#! Inputs:  []
+#! Outputs: []
+export.mem_copy_initial_storage_slots
+  exec.get_native_account_initial_storage_slots_ptr
+  exec.get_native_account_storage_slots_ptr
+  # => [storage_slots_section_ptr, initial_storage_slots_ptr]
+
+  # each slot takes up two words
+  exec.get_num_storage_slots mul.2
+  # => [num_storage_slot_words, storage_slots_section_ptr, initial_storage_slots_ptr]
+
+  exec.mem::memcopy_words
```

Comment:
Would have been cool if we had something like `memcopy_double_words` in the standard library.